std/string_utils.zig

```diff
@@ -0,0 +1,230 @@
+    var moved = true;
+    var index = start;
+
+    while (moved) {
+        moved = false;
+        for (characters) |char| {
+            if (char == a[index]) {
+                moved = true;
+                if (decrement) index -= change else index += change;
+                break;
+            }
+        }
+    }
+    return index;
+}
+
+// If max is 0 then it'll do forever
+pub fn split(a: []const u8, sep: u8, out: &[][]const u8)void {
+    var actualCount: usize = 0;
+    var previousIndex: usize = 0;
+
+    for (a) |char, i| {
+        if (char == sep) {
+            if (i - previousIndex == 0) {
+                (*out)[actualCount] = "";
+            } else {
+                (*out)[actualCount] = a[previousIndex..i];
+            }
+
+            previousIndex = i + 1;
+            actualCount += 1;
+
+            if (actualCount == out.len) break;
+        }
+    }
+
+    (*out)[actualCount] = a[previousIndex..];
+    actualCount += 1;
+    *out = (*out)[0..actualCount];
+}
+
+// Note: characters is an array of u8 not a string!
+// So passing in "abc" doesn't strip abc it strips a, b, and c
+pub fn strip(a: []const u8, characters: []const u8, sides: Side)[]const u8 {
+    var start: usize = 0;
+    var end: usize = a.len - 1;
+    if (sides == Side.LEFT or sides == Side.BOTH) {
+        // Trim left
+        start = impl_strip_side(a, characters, start, 1, false);
+    }
+
+    if (sides == Side.RIGHT or sides == Side.BOTH) {
+        // Trim right
+        end = impl_strip_side(a, characters, end, 1, true);
+    }
+
+    // +1 to convert to 1-index
+    return a[start..end + 1];
+}
+
+pub fn starts_with(a: []const u8, target: []const u8) bool {
+    // Because we are 0-indexing it not 1-indexing it
+    if (a.len < target.len) return false;
+    var i : usize = 0;
+
+    while (i < target.len) {
+        if (a[i] != target[i]) return false;
+        i += 1;
+    }
+    return true;
+}
+
+pub fn ends_with(a: []const u8, target: []const u8) bool {
+    if (a.len < target.len) return false;
+    var diff : usize = a.len - target.len;
+    var i : usize = a.len - 1;
+
+    while (i >= target.len) {
+        if (a[i] != target[i - diff]) return false;
+        i -= 1;
+    }
+    return true;
+}
+
+pub fn str_eql(a: []const u8, b: []const u8) bool {
+    return mem.eql(u8, a, b);
+}
+
+pub fn is_num(byte: u8) bool {
+    return byte >= '0' and byte <= '9';
+}
+
+pub fn to_upper(byte: u8) u8 {
+    return if(is_ascii_lower(byte)) byte - 32 else byte;
+}
+
+pub fn to_lower(byte: u8) u8 {
+    return if(is_ascii_upper(byte)) byte + 32 else byte;
+}
+
+pub fn is_ascii_letter(byte: u8) bool {
+    return is_ascii_lower(byte) or is_ascii_upper(byte);
+}
+
+pub fn is_ascii_lower(byte: u8) bool {
+    return byte >= 'a' and byte <= 'z';
+}
+
+pub fn is_ascii_upper(byte: u8) bool {
+    return byte >= 'A' and byte <= 'Z';
+}
+
+test "String_Utils" {
```

Comment:
I recommend splitting this into some smaller tests like:
```
test "string_utils.is_ascii_letter" {
    assert(is_ascii_letter('C'));
    assert(is_ascii_letter('e'));
    assert(!is_ascii_letter('2'));
}
```

Reply:
Yes of course, I just wanted to get a basic testing suite up first :).